desktop-widgets/divelistview.cpp

```diff
@@ -905,46 +905,53 @@ void DiveListView::loadWebImages()
 	URLDialog urlDialog(this);
 	if (!urlDialog.exec())
 		return;
-	loadImageFromURL(QUrl::fromUserInput(urlDialog.url()));
-}
-
-void DiveListView::loadImageFromURL(QUrl url)
-{
-	if (url.isValid()) {
-		QEventLoop loop;
-		QNetworkRequest request(url);
-		QNetworkReply *reply = manager.get(request);
-		while (reply->isRunning()) {
-			loop.processEvents();
-			sleep(1);
-		}
-		QByteArray imageData = reply->readAll();
-
-		QImage image = QImage();
-		image.loadFromData(imageData);
-		if (image.isNull()) {
-			// If this is not an image, maybe it's an html file and Miika can provide some xslr magic to extract images.
-			// In this case we would call the function recursively on the list of image source urls;
-			report_error(qPrintable(tr("%1 does not appear to be an image").arg(url.toString())));
-			return;
-		}
+	loadImagesFromURLs(urlDialog.url());
+}
+
+void DiveListView::loadImagesFromURLs(QString urls)
+{
+	QStringList validUrls;
+	QStringList urlList = urls.split('\n');
```

Comment:
any reason why you wouldn't want to allow `,` and `;` as separators as well?

Reply:
Laziness, honestly. I went for the cheaper version of split which operates with a char rather than a regular expression. But I have to say: I actually first did this with comma but by accident, the two URLs I tested this with (two random images from the browser) already contained commas in the query string part of the URL. That in retrospect justified for me not splitting on comma (or semicolon for that matter) as well.

Of course I can change this to /[,;\n]/ but do you think it is worth that some URLs that contain these characters then break?